src/Symfony/Component/Lock/Tests/Store/ZookeeperStoreTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Tests\Store;
+
+/*
+ *
+ * @author Ganesh Chandrasekaran <gchandrasekaran@wayfair.com>
+ *
+ * @requires extension zookeeper
+ */
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\Store\ZookeeperStore;
+use Throwable;
+use Zookeeper;
+
+class ZookeeperStoreTest extends AbstractStoreTest
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function setupBeforeClass()
+    {
+        $zookeeper_server = getenv('ZOOKEEPER_HOST').':2181';
+
+        try {
```

Comment:
Does everything really have to be in a try here?

Reply:
Wrapping the logic to get state in this as well in case it throws an exception.

Reply:
_"Since version 0.3.0, this method emits ZookeeperException and it's derivatives."_

For `__construct`: _"This method emits PHP error/warning when parameters count or types are wrong or could not init instance."_ The test should fail when this occurs, not be skipped, otherwise we'll never know if we made a mistake somewhere or if the connection can't be made. However, both cases are a legit reason to fail a test.

For `getState`: _"This method emits PHP error/warning when it fails to get state of zookeeper connection. "_ I would very much like to know if this fails, because it might be a setup issue, I rather not miss this if something broke.

To me it sounds like the whole try/catch should be removed and the `markTestAsSkipped` for the connection underneath the while loop, should trigger a failure instead. If there would be a setup error in travis, we would never be notified of this as everything will always be marked as skipped. This means the test suite will probably just keep reporting that everything is okay.

Reply:
I agree with you on the first 2 points and have removed the `try and catch` blocks. 

However, I feel like connection errors should not be raised as failures because environments can be flaky for many reasons and we should probably have some other way to be monitoring those. I think that's what @jderusse also wanted to do with Redis, Memcached and other stores.

Having said that, how can I install zookeeper extension and run zookeeper server on the Travis?